### PR TITLE
MAINT: Remove some unused imports

### PR DESCRIPTION
**Pull request by jakobjakobson13**

### CODE REVIEW

File: numpy/core/multiarray.py [hunk 1]
```diff
@@ -12,10 +12,9 @@
 from ._multiarray_umath import *  # noqa: F403
 # These imports are needed for backward compatibility,
 # do not change them. issue gh-15518
-# _get_ndarray_c_version is semi-public, on purpose not added to __all__
 from ._multiarray_umath import (
     _fastCopyAndTranspose, _flagdict, _insert, _reconstruct, _vec_string,
-    _ARRAY_API, _monotonicity, _get_ndarray_c_version, _set_madvise_hugepage,
```
Comment (seberg) [inline numpy/core/multiarray.py:18]: Not sure we should remove the first, if it is has a "semi-public" comment.  The second could be removed, but then you need to change the one place where it is used as well (in `__init__.py`).

File: numpy/core/umath.py [hunk 2]
```diff
@@ -10,8 +10,7 @@
 from ._multiarray_umath import *  # noqa: F403
 # These imports are needed for backward compatibility,
 # do not change them. issue gh-11862
-# _ones_like is semi-public, on purpose not added to __all__
-from ._multiarray_umath import _UFUNC_API, _add_newdoc_ufunc, _ones_like
+from ._multiarray_umath import _UFUNC_API, _add_newdoc_ufunc
```
Comment (seberg) [inline numpy/core/umath.py:13]: The comment above specifically asks not to change them.  So a cleanup is not good enough, you can add a comment  if that helps with checkers reporting unused imports.

File: numpy/distutils/mingw32ccompiler.py [hunk 3]
```diff
@@ -15,7 +15,6 @@
 import textwrap
 
 # Overwrite certain distutils.ccompiler functions:
-import numpy.distutils.ccompiler  # noqa: F401
```
Comment (seberg) [inline numpy/distutils/mingw32ccompiler.py:18]: Marked by a "noqa" comment: This import must not be removed.

**Comment by seberg** [comment]
Thanks, but to be honest, I am not sure that any of these changes are actually quite correct.

**Comment by mattip** [comment]
Added a "close" label, not sure that any of these are correct

**Comment by seberg** [comment]
If you are also not sure, I will close this.  Thanks @jakobjakobson13 some cleaups or comments/changes to silence the analyzer (assuming they were found with LGTM or so), would make sense.

**Pull request closed without merging**